nuxeo-liveconnect-core/src/main/java/org/nuxeo/ecm/liveconnect/core/AbstractLiveConnectBlobProvider.java

```diff
@@ -280,15 +282,28 @@ protected Credential getCredential(NuxeoOAuth2Token token) throws IOException {
     }
 
     public final synchronized Credential getCredential(String user) throws IOException {
-        Credential credential = getCredentialFactory().build(user);
-        if (credential == null) {
-            throw new NuxeoException("No credentials found for user " + user + " and service " + blobProviderId);
-        }
-        Long expiresInSeconds = credential.getExpiresInSeconds();
-        if (expiresInSeconds != null && expiresInSeconds.longValue() <= 0) {
-            credential.refreshToken();
+        // suspend current transaction and start a new one responsible for access token update
+        try {
+            return TransactionHelper.runInNewTransaction(() -> {
+                try {
+                    Credential credential = getCredentialFactory().build(user);
+                    if (credential == null) {
+                        throw new NuxeoException(
+                                "No credentials found for user " + user + " and service " + blobProviderId);
+                    }
+                    Long expiresInSeconds = credential.getExpiresInSeconds();
+                    if (expiresInSeconds != null && expiresInSeconds.longValue() <= 0) {
+                        credential.refreshToken();
+                    }
+                    return credential;
```

Comment:
For readability I would put the actual task, which is really the original body of the method, in its own method.